packages/core/src/components/fields/MultiSchemaField.tsx

```diff
@@ -172,11 +170,10 @@ class AnyOfField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends For
     let optionSchema: S;
 
     if (option) {
-      // merge all top level fields except properties
-      const { oneOf, anyOf, properties, ...remaining } = schema;
+      // merge top level required field
+      const { required } = schema;
       // Merge in all the non-oneOf/anyOf properties and also skip the special ADDITIONAL_PROPERTY_FLAG property
-      unset(remaining, ADDITIONAL_PROPERTY_FLAG);
-      optionSchema = !isEmpty(remaining) ? (mergeSchemas(remaining, option) as S) : option;
+      optionSchema = required ? (mergeSchemas({ required }, option) as S) : option;
```

Comment:
Makes sense to me, I can't think of properties other than `required` that should be passed through.